mitxpro/utils.py

```diff
@@ -646,3 +646,57 @@ def strip_datetime(date_str, date_format, date_timezone=None):
 
     date_timezone = date_timezone if date_timezone else datetime.UTC
     return datetime.datetime.strptime(date_str, date_format).astimezone(date_timezone)
+
+
+def validate_courserun_dates(
+    start_date, end_date, enrollment_end, enrollment_start=None, expiration_date=None
+):
+    """
+    Validates course run dates
+
+    Rules:
+    - start_date or enrollment_end must be provided.
+    - start_date and enrollment_end (if provided) must be in the future.
+    - end_date must be later than start_date.
+    - end_date must be in future.
+    - enrollment_end must be later than enrollment_start.
+    - expiration_date must be later than start_date and end_date (if provided).
+
+    Args:
+        start_date (datetime or None): The start date of the course run.
+        end_date (datetime or None): The end date of the course run.
+        enrollment_start (datetime or None): The enrollment start date.
+        enrollment_end (datetime or None): The enrollment end date.
+        expiration_date (datetime or None): The expiration date of the course run.
+
+    Returns:
+        bool: False if validation fails, otherwise True.
+        str or None: An error message if validation fails, otherwise None.
+    """
+    now = now_in_utc()
+    error_msg = None
+
+    if not (start_date or enrollment_end):
+        error_msg = "start_date or enrollment_end must be provided."
```

Comment:
nit: can we store error strings in common place/variables as these are being used in test cases as well.
Minor improvement/changes in error messages can fail test cases as well